packages/fields/types/Content/Implementation.js

```diff
@@ -65,18 +92,36 @@ class Content extends Text {
   extendAdminMeta(meta) {
     return {
       ...meta,
-      // NOTE: We rely on order, which is why we end up with a sparse array
-      blockOptions: this.blocks.map(block => (Array.isArray(block) ? block[1] : undefined)),
+
+      // These are the blocks which have been directly passed in to the Content
+      // field (ie; it doesn't include dependencies unless they too were passed
+      // in)
+      blockTypes: this.blocks.map(block => (Array.isArray(block) ? block[0] : block).type),
+
+      // Key the block options by type to be serialised and passed to the client
+      blockOptions: this.blocks
+        .filter(block => Array.isArray(block) && !!block[1])
+        .reduce(
+          (options, block) => ({
+            ...options,
+            [block[0].type]: block[1],
+          }),
+          {}
+        ),
     };
   }
+
   // Add the blocks config to the views object for usage in the admin UI
   // (ie; { Cell: , Field: , Filters: , blocks: ...})
   extendViews(views) {
     return {
       ...views,
-      blocks: this.blocks.map(block => (Array.isArray(block) ? block[0] : block).viewPath),
+      blocks: unique(
+        flatMap(this.blocks, block => flattenBlockViews(Array.isArray(block) ? block[0] : block))
```

Comment:
Ensuring all the dependency's views are included in this list so they're always bundled even if they're not imported by another block.